### PR TITLE
Problem: proto-gen fails in CI

### LINKED ISSUE #1391: Problem: proto-gen fails in CI
**Describe the bug**

**To Reproduce**

**Expected behavior**

**Screenshots**

**Desktop (please complete the following information):**

**Smartphone (please complete the following information):**

**Additional context**

### PR DESCRIPTION
**Pull request by mmsqe**
👮🏻👮🏻👮🏻 !!!! REFERENCE THE PROBLEM YOUR ARE SOLVING IN THE PR TITLE AND DESCRIBE YOUR SOLUTION HERE !!!! DO NOT FORGET !!!! 👮🏻👮🏻👮🏻

Closes #1391
# PR Checklist:

- [ ] Have you read the [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md)?
- [ ] Does your PR follow the [C4 patch requirements](https://rfc.zeromq.org/spec:42/C4/#23-patch-requirements)?
- [ ] Have you rebased your work on top of the latest master? 
- [ ] Have you checked your code compiles? (`make`)
- [ ] Have you included tests for any non-trivial functionality?
- [ ] Have you checked your code passes the unit tests? (`make test`)
- [ ] Have you checked your code formatting is correct? (`go fmt`)
- [ ] Have you checked your basic code style is fine? (`golangci-lint run`)
- [ ] If you added any dependencies, have you checked they do not contain any known vulnerabilities? (`go list -json -m all | nancy sleuth`)
- [ ] If your changes affect the client infrastructure, have you run the integration test?
- [ ] If your changes affect public APIs, does your PR follow the [C4 evolution of public contracts](https://rfc.zeromq.org/spec:42/C4/#26-evolution-of-public-contracts)?
- [ ] If your code changes public APIs, have you incremented the crate version numbers and documented your changes in the [CHANGELOG.md](https://github.com/crypto-org-chain/chain-main/blob/master/CHANGELOG.md)?
- [ ] If you are contributing for the first time, please read the agreement in [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md) now and add a comment to this pull request stating that your PR is in accordance with the [Developer's Certificate of Origin](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md#developer-certificate-of-originn).

Thank you for your code, it's appreciated! :)

## Summary by CodeRabbit

- **Chores**
    - Enhanced Docker run configuration for building protocols to manage permissions effectively.
    - Updated job name and corresponding make command in workflow for better clarity.
    - Modified workflow comments related to Swagger generation and documentation.

### CODE REVIEW

**Comment by yihuang** [comment]
it only reproduce in #1377

**Comment by mmsqe** [comment]
> it only reproduce in #1377

I thought it was an old issue, seems don't have github.com directory

**Comment by mmsqe** [comment]
@yihuang [0.14.0](https://github.com/orgs/cosmos/packages/container/proto-builder/119928846?tag=0.14.0) seems not work, however works with latest [Dockerfile](https://github.com/cosmos/cosmos-sdk/blob/main/contrib/devtools/Dockerfile)

File: Makefile [hunk 1]
```diff
@@ -272,7 +272,7 @@ endif
 HTTPS_GIT := https://github.com/crypto-org-chain/cronos.git
 protoVer=0.14.0
 protoImageName=ghcr.io/cosmos/proto-builder:$(protoVer)
-protoImage=$(DOCKER) run --rm -v $(CURDIR):/workspace --workdir /workspace $(protoImageName)
+protoImage=$(DOCKER) run --rm -v $(CURDIR):/workspace --workdir /workspace --user root $(protoImageName)
```
Comment (yihuang) [inline Makefile:275]: awsome, but I guess we need distinguish local run and CI run?

**Review by yihuang — APPROVED** [review]

**Pull request merged**